src/mcse.jl

```diff
@@ -94,7 +94,7 @@ function _mcse(
 end
 
 function _mcse_quantile(x, p, Seff)
-    Seff === missing && return missing
+    (Seff === missing || isnan(Seff)) && return Seff
```

Comment:
Given that the other branch returns `(x[upper] - x[lower])/2`, it seems this could easily lead to type instabilities? I think something like

```suggestion
    Seff === missing && return missing
    if isnan(Seff)
        T = eltype(x) / 1
        return T(NaN)
    end
```

would be safer.